apps/web/src/components/(teams)/dialogs/invite-team-member-dialog.tsx

```diff
@@ -70,41 +74,61 @@ export const InviteTeamMembersDialog = ({
   ...props
 }: InviteTeamMembersDialogProps) => {
   const [open, setOpen] = useState(false);
+  const uploadRef = useRef<HTMLInputElement | null>(null);
+  const [invitationType, setInvitationType] = useState('INDIVIDIUAL');
 
   const { toast } = useToast();
 
   const form = useForm<TInviteTeamMembersFormSchema>({
     resolver: zodResolver(ZInviteTeamMembersFormSchema),
     defaultValues: {
-      invitations: [
-        {
-          email: '',
-          role: TeamMemberRole.MEMBER,
-        },
-      ],
+      email: '' || [''],
+      role: TeamMemberRole.MEMBER,
     },
   });
 
-  const {
-    append: appendTeamMemberInvite,
-    fields: teamMemberInvites,
-    remove: removeTeamMemberInvite,
-  } = useFieldArray({
-    control: form.control,
-    name: 'invitations',
-  });
-
   const { mutateAsync: createTeamMemberInvites } = trpc.team.createTeamMemberInvites.useMutation();
 
-  const onAddTeamMemberInvite = () => {
-    appendTeamMemberInvite({
-      email: '',
-      role: TeamMemberRole.MEMBER,
-    });
+  const handleFileUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.files?.length) {
+      return;
+    }
+    const file = e.target.files[0];
+
+    if (file) {
+      const reader = new FileReader();
+      const emailRegex = /^([A-Z0-9_+-]+\.?)*[A-Z0-9_+-]@([A-Z0-9][A-Z0-9-]*\.)+[A-Z]{2,}$/i;
+      reader.onload = (e) => {
```

Comment:
This is a bit scary, things like papaparse and such exist for handling the reading of CSV files in a safe manner since the format itself is prone to weirdness.

Reply:
Thanks for the suggestion. Papaparse would be good for handling the csv file. Let me quickly add that.

Reply:
Consider the bundle size implications as well there may be smaller and just as good options 😄 Papaparse was the most popular last I messed with CSV's but I'm unsure as to what current recommendations are.

https://bundlephobia.com/ is a fantastic website for seeing how big something might be!

Reply:
Okay😄. Let me check this out.